File: Image_Compression/sptl2.m
function [pred11] = sptl2(D)
const=10;
[row,col]=size(D);
pred1=zeros(row+1,col+1);
pred11=zeros(row+1,col+1);

pred1(1,:)=const*ones(1,col+1);
pred1(2:row+1,1)=const*ones(row,1);

pred1(2:row+1,2:col+1)=D;

for i=2:row+1
  for j=2:col+1    
        pred11(i,j)=0.5*(pred1(i,j-1)+pred1(i-1,j));
  end
end

pred11=pred11(2:row+1,2:col+1);
pred11=D-fix(pred11);

File: Image_Compression/sptl2.py
import numpy as np

def sptl2(D):
    const = 5
    row = D
    col = D
    pred1 = np.zeros((D, D))
    pred11 = np.zeros((D, D))

    pred1[0, :] = const
    #print(pred1)

    pred1[1:row, 0] = const
    #print(pred1)

    pred1[1:row, 1:col] = D
    print(pred1)
    print(pred11)

    #X = (A+B)/2
    for i in range(1, row):
        for j in range(1, col):
            pred11[i][j] = 0.5 * (pred1[i][j-1] + pred1[i-1][j])
    
    pred11 = pred11[1:row, 1:col]
    print(pred11)
    pred11 = D - np.fix(pred11)
    print(pred11)
    pred11 = pred11.astype('double')
    print(pred11)